packages/protocol-http/src/extensions/httpExtensionConfiguration.ts

```diff
@@ -0,0 +1,42 @@
+import { HttpHandler } from "../httpHandler";
+import {
+  getHttpHandlerConfiguration,
+  resolveHttpHandlerRuntimeConfig as _resolveHttpHandlerRuntimeConfig,
+} from "./http";
+
+/**
+ * @internal
+ */
+export interface HttpHandlerExtensionConfiguration<HandlerConfig extends object = Record<string, unknown>> {
```

Comment:
I think we can remove the generic here.

Reply:
I was discussing it with @AndrewFossAWS and we found a part of the spec that says the config "bag" should be treated in a type-safe manner if possible, so I believe that supports using this generic here. It has a fallback default that makes it easier to use if the config type is not known.